### PR TITLE
ref(js): Convert GroupEvents to a FC

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/views/issueDetails/groupEvents.tsx [hunk 1]
```diff
@@ -33,86 +25,59 @@ const excludedTags = [
...
+    () =>
+      browserHistory.listen(newLocation =>
+        handleRouteLeave({
+          fieldsToClean: ['cursor'],
+          newLocation,
+          oldPathname: location.pathname,
+        })
+      ),
+    [location]
+  );
```
Comment (scttcper) [inline static/app/views/issueDetails/groupEvents.tsx:41]: can we use `useCleanQueryParamsOnRouteLeave` here

Comment (evanpurkhiser) [reply static/app/views/issueDetails/groupEvents.tsx:41]: Is that a "please use this" or is it a "I dunno if it will work but maybe"

**Review by scttcper — APPROVED** [review]

File: static/app/views/issueDetails/groupEvents.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/views/issueDetails/groupEvents.tsx:41]: i haven't used it yet but i know it exists. also use it

Comment (evanpurkhiser) [reply static/app/views/issueDetails/groupEvents.tsx]: @ryan953 did I use this rgiht

Comment (evanpurkhiser) [reply static/app/views/issueDetails/groupEvents.tsx]: Go add more comments :-)

File: static/app/views/issueDetails/groupEvents.tsx [hunk 2]
```diff
@@ -33,86 +25,51 @@ const excludedTags = [
...
 
-class GroupEvents extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    const queryParams = this.props.location.query;
-
-    this.state = {
-      query: queryParams.query || '',
-    };
```
Comment (ryan953) [inline static/app/views/issueDetails/groupEvents.tsx:44]: Why was this.?!?! ?

**Review by ryan953 — APPROVED** [review]

**Pull request merged**